sdk/servicebus/Microsoft.Azure.ServiceBus/src/QueueClient.cs

```diff
@@ -151,6 +151,54 @@ public QueueClient(ServiceBusConnection serviceBusConnection, string entityPath,
             MessagingEventSource.Log.QueueClientCreateStop(serviceBusConnection.Endpoint.Authority, entityPath, this.ClientId);
         }
 
+        /// <summary>
+        /// Creates a new instance of the Queue client using Azure Active Directory authentication.
+        /// </summary>
+        /// <param name="endpoint">Fully qualified domain name for Service Bus. Most likely, {yournamespace}.servicebus.windows.net</param>
+        /// <param name="entityPath">Queue path.</param>
+        /// <param name="authCallback">User provided delegate that will provide the access token.</param>
+        /// <param name="authority">Address of the authority to issue token.</param>
+        /// <param name="transportType">Transport type.</param>
+        /// <param name="receiveMode">Mode of receive of messages. Defaults to <see cref="ReceiveMode"/>.PeekLock.</param>
+        /// <param name="retryPolicy">Retry policy for queue operations. Defaults to <see cref="RetryPolicy.Default"/></param>
+        /// <remarks>Creates a new connection to the queue, which is opened during the first send/receive operation.</remarks>
+        public static QueueClient CreateWithAzureActiveDirectory(
```

Comment:
>CreateWithAzureActiveDirectory [](start = 34, length = 30)

1. Do we need this API? We already have the overload where you can pass your own ITokenProvider.

Reply:
We also accept `connectionStringBuilder`. So lets not duplicate the APIs

---
In reply to: [286696363](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r286696363) [](ancestors = 286696363)

Reply:
I'm curious as to the purpose of not just documenting that the consumer should create and use an `CreateAzureActiveDirectoryTokenProvider` or `CreateAzureActiveDirectoryTokenProvider` and pass them to the `ITokenProvider` overload.   Is there a benefit to consumers that I'm overlooking?

Reply:
I don't want to block on this, but I'm still curious as to the answer.  I feel that I'm missing some important context...

Reply:
It's removed and will no longer part of this change. It was an effort to keep consistency with the EventHubs SDK but it seems like there are good reasons to avoid these redundant APIs.

---
In reply to: [292678097](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r292678097) [](ancestors = 292678097)